mesonbuild/wrap/wrap.py

```diff
@@ -525,8 +543,24 @@ def get_data(self, urlstring: str) -> T.Tuple[str, str]:
         elif WHITELIST_SUBDOMAIN in urlstring:
             raise WrapException(f'{urlstring} may be a WrapDB-impersonating URL')
         else:
+            headers = {}
+            is_http_protocol = url.scheme in ('http', 'https')
+            creds = self.get_netrc_credentials(url.netloc)
+
+            if is_http_protocol:
+                headers.update({'User-Agent': f'mesonbuild/{coredata.version}'})
+
+            req_urlstring = urlstring
+            if creds is not None and '@' not in url.netloc:
+                login, password = creds
+                if is_http_protocol:
+                    enc_creds = b64encode(f'{login}:{password}'.encode()).decode()
+                    headers.update({"Authorization": f"Basic {enc_creds}"})
```

Comment:
digest auth would be preferable to basic auth, if possible.

Reply:
My understanding, but I could be wrong, is digest is mostly useful when used with plain text because it hash the actual password. Since we are restricting this to https, not sure it's needed, is it?

Reply:
Yeah, if digest doesn't actually result in a working feature then I guess we cannot use it. Sad, but it is what it is.

We do definitely need to restrict it to no http, in that case.

Reply:
Nitpicking: `f"Basic {enc_creds}"` we use single quote strings by convention in Meson.